### ISSUE TITLE
Impossible to overfit even on 10 sample

### ISSUE THREAD

**Issue by botcs** [matterport/Mask_RCNN#826]
Hi,

I have written a training script using the balloons.py and the coco.py following the instructions on Medium.
Using the data inspection notebooks I have double-checked every sample in this small dataset (10 images, all classes are also present in COCO, angle is different).

If I use my own indices, total in 6 class + background then I have to remove the last mrcnn layers - which results in catastrophic forgetting.
I have experimented with training only the last layers of the mask_rcnn and adding layer after layer to the trainable params, eventually adding RPN to the trainable set as well, resulting in the "heads" preset used in the balloon experiment. Finally I have tried training all trainable parameters using "all" preset, with no luck.
Tried reducing the learning rate, switching to Adam... Still the results show that if I change anything from the COCO weights than it crashes.

I tried training on the same mini dataset with using the complete set of COCO weights and reindexing my classes to match the IDs in COCO annotations, therefore the shape of the last layers could be restored as well.
It performs "OK", which means only a very little difference from the original COCO weights and insignificant improvement.
Usually the losses reported by the script is the following:
`loss: 0.6037 - rpn_class_loss: 0.3280 - rpn_bbox_loss: 0.2757 - mrcnn_class_loss: 6.7830e-06 - mrcnn_bbox_loss: 0.0000e+00 - mrcnn_mask_loss: 0.0000e+00 - val_loss: 0.4442 - val_rpn_class_loss: 0.3250 - val_rpn_bbox_loss: 0.1192 - val_mrcnn_class_loss: 4.7684e-06 - val_mrcnn_bbox_loss: 0.0000e+00 - val_mrcnn_mask_loss: 0.0000e+00`

On a larger dataset of 7k images the loss converges to a local minima, but visual output is completely wrong, barely finding any object, even on the training samples.

What should be the next step, where did I miss something?

Thank you,
Csabi

**Comment by botcs** [comment]
FIX: don't use `pad64` ... it was a long debugging session, but I have learned that both crop and square resize methods are working. Maybe it has to do something with not rectangular image shapes
